insights/insightsclient/insightsclient.go

```diff
@@ -27,57 +27,67 @@ import (
 	configv1 "github.com/openshift/api/config/v1"
 	apimachineryversion "k8s.io/apimachinery/pkg/version"
 
-	"github.com/openshift/insights-operator/pkg/authorizer"
+	"github.com/redhatinsights/insights-ingress-http-client/authorizer"
 )
 
 const (
 	responseBodyLogLen = 1024
 )
 
+// Client The client structure for making requests to cloud.redhat.com
 type Client struct {
-	client      *http.Client
-	maxBytes    int64
-	metricsName string
+	client       *http.Client
+	maxBytes     int64
+	metricsName  string
+	operatorName string
+	mimeType     string
 
 	authorizer  Authorizer
 	clusterInfo ClusterVersionInfo
 }
 
+// Authorizer An interface for adding authorization and proxy information the request
 type Authorizer interface {
 	Authorize(req *http.Request) error
 	NewSystemOrConfiguredProxy() func(*http.Request) (*url.URL, error)
 }
 
+// ClusterVersionInfo An interface for the cluster version object
 type ClusterVersionInfo interface {
 	ClusterVersion() *configv1.ClusterVersion
 }
 
+// Source An object for storing data of multiple types
 type Source struct {
 	ID       string
 	Type     string
 	Contents io.Reader
 }
 
+// ErrWaitingForVersion An error due to cluster version responding slowly
 var ErrWaitingForVersion = fmt.Errorf("waiting for the cluster version to be loaded")
 
-func New(client *http.Client, maxBytes int64, metricsName string, authorizer Authorizer, clusterInfo ClusterVersionInfo) *Client {
+// New Initialize a new client object
+func New(client *http.Client, maxBytes int64, metricsName string, operatorName string, mimeType string, authorizer Authorizer, clusterInfo ClusterVersionInfo) *Client {
 	if client == nil {
 		client = &http.Client{}
 	}
 	if maxBytes == 0 {
 		maxBytes = 10 * 1024 * 1024
 	}
 	return &Client{
-		client:      client,
-		maxBytes:    maxBytes,
-		metricsName: metricsName,
-		authorizer:  authorizer,
-		clusterInfo: clusterInfo,
+		client:       client,
+		maxBytes:     maxBytes,
+		metricsName:  metricsName,
+		operatorName: operatorName,
+		mimeType:     mimeType,
+		authorizer:   authorizer,
+		clusterInfo:  clusterInfo,
 	}
 }
 
 func getTrustedCABundle() (*x509.CertPool, error) {
-	caBytes, err := ioutil.ReadFile("/var/run/configmaps/trusted-ca-bundle/ca-bundle.crt")
+	caBytes, err := ioutil.ReadFile("/var/run/configmaps/trusted-ca-bundle/ca-bundle.crt") //Should this be parameterized?
```

Comment:
I think this should probably be parameterized, but I forget if this is just a standard location. Either way we should open an issue to capture that the trusted-ca cert needs to be in place for the library.